integration_tests/test_domestic_migration.py

```diff
@@ -123,3 +123,88 @@ def test_only_living_people_move(simulants_on_adjacent_timesteps):
+
+# add employer address tests
+# 2. all simulants with same employer id have same employer details (static)
+# 3. all employer ids have unique address details (static)
+
+
+def test_addresses_during_moves(simulants_on_adjacent_timesteps):
+    """Check that address details change after a move. Note that
+    this test does not distinguish between the different types of moves (eg
+    household vs individual moves)
+    """
+    unit_id_col = "employer_id"
+    address_id_col = "business_details.employer_address_id"
+    other_details_cols = ["business_details.employer_name"]
+    _test_addresses_during_moves(
+        pops=simulants_on_adjacent_timesteps,
+        unit_id_col=unit_id_col,
+        address_id_col=address_id_col,
+        other_details_cols=other_details_cols,
+    )
+
+
+def _test_addresses_during_moves(pops, unit_id_col, address_id_col, other_details_cols):
+    details_cols = [unit_id_col, address_id_col] + other_details_cols
+    for before, after in pops:
+        mask_movers = before[address_id_col] != after[address_id_col]
+        # address details change if address_id changes
+        # NOTE: we cannot assert that all details are different because some
+        # units (eg employer "not implemented") have the same name.
+        assert (
+            not (
+                before.loc[mask_movers, [unit_id_col] + details_cols]
+                == after.loc[mask_movers, [unit_id_col] + details_cols]
+            )
+            .all()
+            .all()
+        )
+        # address details do not change if address_id does not change
+        pd.testing.assert_frame_equal(
+            before.loc[~mask_movers, details_cols], after.loc[~mask_movers, details_cols]
+        )
+
+        # TODO when puma/state is implemented
+        # # TODO Check that *most* are in a new state when we add all locations
+        # # Some movers are in a new state.
+        # # assert any(before.loc[mask_movers, "state"] != after.loc[mask_movers, "state"])
+
+        # # Most movers are in a different puma
+        # assert (
+        #     before.loc[mask_movers, "puma"] != after.loc[mask_movers, "puma"]
+        # ).sum() / mask_movers.sum() >= 0.95
+
+
+def test_address_uniqueness(populations):
+    """Check that all units (households or employers) have unique details
+    and also that simulants of the same unit have the same details
+    """
+    unit_id_col = "employer_id"
+    address_id_col = "business_details.employer_address_id"
+    other_details_cols = ["business_details.employer_name"]
+    _test_address_uniqueness(
+        pops=populations,
+        unit_id_col=unit_id_col,
+        address_id_col=address_id_col,
+        other_details_cols=other_details_cols,
+    )
+
+
+def _test_address_uniqueness(pops, unit_id_col, address_id_col, other_details_cols):
+    details_cols = [unit_id_col, address_id_col] + other_details_cols
+    for pop in pops:
+        assert (pop.groupby(unit_id_col)[details_cols].nunique() == 1).all().all()
```

Comment:
This is testing that everyone with the same employer has the same employer address. Aren't you also supposed  supposed to test that each employer has a different address?